src/cli/subdoc_get.rs

```diff
@@ -0,0 +1,530 @@
+                None,
+            )
+            .named(
+                "bucket",
+                SyntaxShape::String,
+                "the name of the bucket",
+                None,
+            )
+            .named("scope", SyntaxShape::String, "the name of the scope", None)
+            .named(
+                "collection",
+                SyntaxShape::String,
+                "the name of the collection",
+                None,
+            )
+            .named(
+                "databases",
+                SyntaxShape::String,
+                "the databases which should be contacted",
+                None,
+            )
+            .named(
+                "batch-size",
+                SyntaxShape::Number,
+                "the maximum number of items to batch send at a time",
+                None,
+            )
+            .switch("halt-on-error", "halt on any errors", Some('e'))
+            .category(Category::Custom("couchbase".to_string()))
+    }
+
+    fn usage(&self) -> &str {
+        "Fetches the value of the provided path in the specified document through the data service"
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        match call.req::<Value>(engine_state, stack, 0)? {
+            Value::String { val: _, span: _ } => {
+                run_subdoc_get(self.state.clone(), engine_state, stack, call, input)
+            }
+            Value::List { vals: _, span: _ } => {
+                run_subdoc_multi_lookup(self.state.clone(), engine_state, stack, call, input)
+            }
+            _ => {
+                return Err(ShellError::GenericError(
+                    "Please supply a single field as a string or multiple as a list`".to_string(),
+                    "".to_string(),
+                    None,
+                    None,
+                    Vec::new(),
+                ));
+            }
+        }
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example{
+                description: "Fetches the address field from the document with the ID landmark_10019",
+                example: "subdoc get address landmark_10019",
+                result: None
+            },
+            Example{
+                description: "Fetches the address and content fields from the document with the ID landmark_10019",
+                example: "subdoc get [address content] landmark_10019",
+                result: None
+            },
+            Example{
+                description: "Fetches address and content fields from multiple documents with IDs from the previous command",
+                example: "[landmark_10019, landmark_10020] | subdoc get [address, content]",
+                result: None
+            },
+        ]
+    }
+}
+
+fn run_subdoc_get(
```

Comment:
Is there a way we can DRY these 2 calls up a bit? They're almost identical aside from the op types.

Reply:
Done